src/rules/converters/space-before-function-paren.ts

```diff
@@ -1,11 +1,22 @@
 import { RuleConverter } from "../converter";
 
+function isExistedESLintOption(rule: string) {
+    return ["anonymous", "named", "asyncArrow"].includes(rule);
+}
+
 export const convertSpaceBeforeFunctionParen: RuleConverter = tslintRule => {
+    const tslintRuleArguments = tslintRule.ruleArguments;
+    const ruleArguments = tslintRuleArguments.filter(isExistedESLintOption);
     return {
         rules: [
             {
-                ...(tslintRule.ruleArguments.length !== 0 && {
-                    ruleArguments: tslintRule.ruleArguments,
+                ...(tslintRuleArguments.length !== 0 && {
+                    ruleArguments,
+                    ...(tslintRuleArguments.length > ruleArguments.length && {
+                        notices: [
+                            'ESLint doesn\'t support the following options: "method" and "constructor". These two options can be covered by "named" therefore make sure you have set this option.',
```

Comment:
A couple of things:

* It would be a little confusing to have only one of the two unsupported options enabled in TSLint, then get a message about the other one in your log. Maybe this should only mention the option(s) you have enabled?
* I don't follow what you mean by `covered by "named"`. Is that meant to be a helpful hint that the equivalent ESLint rule option to enable is `named`? If so, is the proper output of the converter when both those TSLint options are enabled to have that ESLint option turned on?

Reply:
The first point is definitely reasonable. I will collect notices properly.

Regarding the second point, I would say I've provided not ideal description `covered by "named"`. _TSLint's_ `"named"` !== _ESLint's_ `"named"`, but `eslint` includes `"constructor"` and `"method"` into the `"named"` option (that's what I mean by `covered by "named"`). Here some examples explaining the difference:

 - tslint [`"space-before-function-paren": [true, { "named": "always" }]`](https://palantir.github.io/tslint-playground/?saved=N4Igxg9gJgpiBcIBmBXAdmALgSwmgBEhBPgBQCU+wAOgfvQE4yYoMGYMowDctAvrVpgANgEMAzuPwAxYlVr16AeiX40JGAwYQGUmNkwALTfkhpxHFFh34bAW2aHo8BYrMXO1hhXrB8SgCoA1QE6egAHFAAjYWwwfAcjaB8-QOD8UNCQABpwPCRsAHMEEBow-GoQGAAPTBg0KHFK+HwAbUrMcVi0THgmSDsHBpgoSoBdbNd6Ss5hGCaEeXLFCpBxcNEwGABaKJgiJm3UDBw8bY2mNGa2yxhsqlW0UQdRxcrRYQB3UQBPBYyxlMMvxaCA+EA) vs. eslint [`space-before-function-paren: [2, { "named" : "always" }]`](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgc3BhY2UtYmVmb3JlLWZ1bmN0aW9uLXBhcmVuOiBbMiwgeyBcIm5hbWVkXCIgOiBcImFsd2F5c1wiIH1dKi9cbi8qZXNsaW50LWVudiBlczYqL1xuXG5mdW5jdGlvbiBmb28gKCkge31cblxuY2xhc3MgRm9vIHtcbiAgICBjb25zdHJ1Y3RvcigpIHt9XG4gICAgaGVsbG8oKSB7fVxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTAsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)
 - tslint [`"space-before-function-paren": [true, { "constructor": "always", "method": "always" }]`](https://palantir.github.io/tslint-playground/?saved=N4Igxg9gJgpiBcIBmBXAdmALgSwmgBEhBABQCU+wAOgfnQE4yYr0Gb0owDcNAvjTTAAbAIYBnMfgBixSjTp1IaMexRYI9fOTrB8AegBUBvfn606ABxQAjIdjD40IgLYwoWirsPHTfGiAAacDwkbABzBBBqc3wqEBgAD0wYNCgxOPh8AG04zDE7NEx4RkhnV1S3OIBdAPkFOI4hGHSEORiFWJAxCxEwGABaaxgiRn7UDBw8fp7GNAzs1RgAyk6lFQ51enm4kSEAdxEATxbluNdMAAtobZBdg+O40yq6ujMzEF4gA) vs. eslint [`space-before-function-paren: [2, { "named" : "always" }]`](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgc3BhY2UtYmVmb3JlLWZ1bmN0aW9uLXBhcmVuOiBbMiwgeyBcIm5hbWVkXCIgOiBcImFsd2F5c1wiIH1dKi9cbi8qZXNsaW50LWVudiBlczYqL1xuXG5mdW5jdGlvbiBmb28oKSB7XG4gICAgcmV0dXJuIHRydWU7XG59XG5cbmNsYXNzIEZvbyB7XG4gICAgY29uc3RydWN0b3IgKCkgIHsgLyoqLyB9XG4gICAgbmFtZWQgKCkgeyAvKiovIH1cbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEwLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)


Do you have ideas on how to make the notice to be simple and understandable at the same time?

Reply:
What you have now with ` 'Option "constructor" is not supported by ESLint.'` seems nice and simple. I'm pretty mediocre at thinking of these styles, so let's go with that! 🚀